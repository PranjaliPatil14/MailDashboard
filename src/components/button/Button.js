import React from "react";
import * as PropTypes from "prop-types";
import "./button.css";

const Button = ({ children, onClick, className, testId }) => {
  return (
    <button
      type="submit"
      onClick={onClick}
      className={`button ${className}`}
      data-testid={testId}
    >
      {children}
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func,
  className: PropTypes.string,
  testId: PropTypes.string,
};

Button.defaultProps = {
  className: "",
  onClick: () => {},
  testId: "",
};

export default Button;
