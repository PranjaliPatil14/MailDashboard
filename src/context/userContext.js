import React, { useEffect, useState } from "react";
import * as PropTypes from "prop-types";
import useLocalStorage, { USER } from "../utilities/useLocalStorage";

const initialUserContextValue = {
  users: [],
  loggedInUser: {
    id: 0,
    name: "",
    role: "",
    profileImage: "",
  },
  setLoggedInUser: () => {},
  setUsers: () => {},
};

export const UserContext = React.createContext(initialUserContextValue);

const UserContextProvider = ({ children }) => {
  const [users, setUsers] = useState([]);
  const [loggedInUser, setLoggedInUser] = useState({});
  const { value } = useLocalStorage(USER);

  useEffect(() => {
    if (!users.length) {
      setUsers(value);
    }
  }, [users.length, value]);

  return (
    <UserContext.Provider
      value={{ users, setUsers, loggedInUser, setLoggedInUser }}
    >
      {children}
    </UserContext.Provider>
  );
};

UserContextProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default UserContextProvider;
