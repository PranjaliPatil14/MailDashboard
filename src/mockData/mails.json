[
  {"id": 1, "from":1,"to":[3],"cc":[2],"body":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla eget rhoncus metus.","attachment":false,"time":"Sat Jan 02 2021 18:02:02 GMT+0530 (India Standard Time)","tag":"Development", "read": false, "important": false},
  {"id": 2, "from":2,"to":[1],"cc":[3],"body":"Quisque non lectus tincidunt, consequat sapien eu, rutrum nisl.","attachment":false,"time":"Tue Jan 05 2021 07:44:34 GMT+0530 (India Standard Time)","tag":"Jobs", "read": false, "important": false},
  {"id": 3, "from":1,"to":[3],"cc":[],"body":"Phasellus ullamcorper nisl a velit vulputate, in euismod felis sollicitudin.","attachment":false,"time":"Sat Jan 02 2021 18:02:02 GMT+0530 (India Standard Time)","tag":"", "read": false, "important": false},
  {"id": 4, "from":2,"to":[1],"cc":[],"body":"Integer lacinia, ex vitae interdum lobortis, felis nisi pretium tortor, et tincidunt dolor elit sed magna.","attachment":true,"time":"Sat Jan 02 2021 18:02:02 GMT+0530 (India Standard Time)","tag":"client", "read": true, "important": true},
  {"id": 5, "from":3,"to":[2],"cc":[],"body":"Integer lacinia, ex vitae interdum lobortis, felis nisi pretium tortor, et tincidunt dolor elit sed magna.","attachment":true,"time":"Sat Jan 02 2021 18:02:02 GMT+0530 (India Standard Time)","tag":"", "read": true, "important": true},
  {"id": 6, "from":2,"to":[3],"cc":[],"body":"Integer lacinia, ex vitae interdum lobortis, felis nisi pretium tortor, et tincidunt dolor elit sed magna.","attachment":true,"time":"Sat Jan 02 2021 18:02:02 GMT+0530 (India Standard Time)","tag":"", "read": true, "important": true},
  {"id": 7, "from":3,"to":[1],"cc":[2],"body":"Integer lacinia, ex vitae interdum lobortis, felis nisi pretium tortor, et tincidunt dolor elit sed magna.","attachment":true,"time":"Sat Jan 02 2021 18:02:02 GMT+0530 (India Standard Time)","tag":"", "read": true, "important": true}
]
