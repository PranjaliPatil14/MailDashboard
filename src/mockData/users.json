[{"id": 1, "name": "abc", "role": "Art director", "email": "abc@gmail.com", "password": "abc"},{"id": 2, "name": "pqr", "role": "ABC Director", "email": "pqr@gmail.com", "password": "pqr"},{"id": 3, "name": "mat", "role": "PQA director", "email": "mat@gmail.com","password": "mat"}]
