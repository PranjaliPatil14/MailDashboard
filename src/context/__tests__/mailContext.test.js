import React, { useContext } from "react";
import { render, fireEvent } from "@testing-library/react";
import Button from "../../components/button/Button";
import mails from "../../mockData/mails.json";
import users from "../../mockData/users.json";
import MailContextProvider, { MailContext } from "../mailContext";
import { UserContext } from "../userContext";
import { MARK_DELETED } from "../mailReducer";

const MockComponent = () => {
  const { mails: value, loggedInUserMails, dispatch } = useContext(MailContext);
  return (
    <>
      <span>{value[1]?.body}</span>
      <span>{loggedInUserMails[0]?.body}</span>
      <Button onClick={() => dispatch({ type: MARK_DELETED, data: [1] })}>
        delete
      </Button>
    </>
  );
};

describe("userContext", () => {
  test("should pass mails, loggedInUserMails, dispatch props to child component wrapped in mail Context provider", () => {
    Storage.prototype.getItem = jest.fn(() => JSON.stringify(mails));
    Storage.prototype.setItem = jest.fn();

    const loggedInUser = users[1];
    const { asFragment, getByText, queryByText } = render(
      <UserContext.Provider value={{ loggedInUser }}>
        <MailContextProvider>
          <MockComponent />
        </MailContextProvider>
      </UserContext.Provider>
    );

    expect(asFragment()).toMatchSnapshot();
    expect(Storage.prototype.setItem).toHaveBeenCalled();

    fireEvent.click(getByText("delete"));

    expect(
      queryByText(
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla eget rhoncus metus."
      )
    ).not.toBeInTheDocument();
  });
});
