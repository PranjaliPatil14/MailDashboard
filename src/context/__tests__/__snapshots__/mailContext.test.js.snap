// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userContext should pass mails, loggedInUserMails, dispatch props to child component wrapped in mail Context provider 1`] = `
<DocumentFragment>
  <span>
    Quisque non lectus tincidunt, consequat sapien eu, rutrum nisl.
  </span>
  <span>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla eget rhoncus metus.
  </span>
  <button
    class="button "
    data-testid=""
    type="submit"
  >
    delete
  </button>
</DocumentFragment>
`;
